### ISSUE TITLE
[IOS] appStoreReceipt not correct

### ISSUE THREAD

**Issue by GrindingOgre** [j3k0/cordova-plugin-purchase#1222]
```
My environment:
OSX Catalina 10.15.7 (19H15)
Cordova 9.0.0
Device: All iphones
Plugin Version: 5.1.1
```

### Expected behavior

When i receive the appStoreReceipt and contact apple to validate a paid subscription, it should give me all the detail i need.

### Observed behavior

When i send the receipt to https://buy.itunes.apple.com/verifyReceipt, Apple tells me that the receipt is malformed.

### Steps to reproduce

```
{
	"id": "*****",
	"alias": "****",
	"type": "paid subscription",
	"group": "20835144",
	"state": "approved",
	"title": "*****",
	"description": "****",
	"priceMicros": ****,
	"price": "****",
	"currency": "EUR",
	"countryCode": "IT",
	"loaded": true,
	"canPurchase": false,
	"owned": true,
	"introPrice": null,
	"introPriceMicros": null,
	"introPricePeriod": null,
	"introPricePeriodUnit": null,
	"introPricePaymentMode": null,
	"ineligibleForIntroPrice": null,
	"discounts": [],
	"downloading": false,
	"downloaded": false,
	"additionalData": null,
	"transaction": {
		"type": "ios-appstore",
		"appStoreReceipt": "MIIUOAYJK...dh14o4aw="
	},
	"billingPeriod": 3,
	"billingPeriodUnit": "Month",
	"valid": true
}
```

This product object has wrong appStoreReceipt.

I've attached the apple store on fovea (to test it) following the instructions and it sent me the correct appStoreReceipt (if i call https://buy.itunes.apple.com/verifyReceipt, it returns me the right data).

Is there a problem with the base64 encoding? My app is in production and people on ios are subscribing without getting the premium features.

**Comment by Dexus** [comment]
did you set the secret on fovea?

**Comment by j3k0** [comment]
Are you really using plugin version `5.1.1`? The latest version is `10.6.1`, if so I highly advise you to upgrade. You need to send to Apple the content of `transaction.appStoreReceipt`. The content looks correct, you're probably not calling the endpoint correctly.

Did it all work correctly with Fovea's service https://billing.fovea.cc ?

**Comment by GrindingOgre** [comment]
I wrote the plugin version of cordova-ios, but cordova-plugin-purchase in my app is at version 10.5.3.
I couldn't try in production in Fovea's service, but i think i found the culprit.

The fact is that we use JSON.stringify to send all the product objects to our server (written in PHP and our webservices can accept only application/x-www-form-urlencoded); in this serialization, i think we lost the correct appStoreReceipt.
We tried with a testflight and it worked well by sending the appStoreReceipt directly to our servers.
